hax/hax/ha/events.py

```diff
@@ -89,13 +88,17 @@ def __init__(self,
     def _subscribe(self, event_list: List[SubscribeEvent]) -> str:
         logging.info('Subscribing to events: %s', event_list)
         # TODO create a PR for cortx-ha. The type annotation seems to be wrong
-        kafka_topic_name = str(
-            self.event_manager.subscribe(COMPONENT_ID, event_list))
-        return kafka_topic_name
+
+        # HA functionality won't work in containers. Hare is the only
+        # subscriber to HA's event manager, and the topic name is hardcoded,
+        # in HA code. So, topic prefix is imported which is in format
+        # 'ha_event_<component_id>'
+        ha_topic_prefix = str(EVENT_MANAGER_KEYS.MESSAGE_TYPE_VALUE.value)
+        topic = ha_topic_prefix.replace("<component_id>", COMPONENT_ID)
```

Comment:
It looks as if we're re-implementing the logic that should be in cortx-ha. Isn't it strange?

Reply:
@Shreya-18, I think earlier api was better that the component provides the COMPONENT_ID and cortx-ha prepares the topic and returns the same. Now, this is distributed between the component and cortx-ha.
The subscribe, unsubscribe methods were actually better, any idea why were they replaced with this implementation? I think number of users must not matter here.

Reply:
@mssawant, @knekrasov, with the earlier api, we received an instance of `EventManager()`, for calling subscribe() and unsubscribe() methods.
`self.event_manager = EventManager.get_instance()` -- for this, we need HA running (configured by mini-prov). else it threw below error.
```
File "/usr/lib/python3.6/site-packages/ha/core/event_manager/event_manager.py", line 54, in get_instance
    EventManager(default_log_enable, singleton_check=True)
  File "/usr/lib/python3.6/site-packages/ha/core/event_manager/event_manager.py", line 70, in __init__
    ConfigManager.init(const.EVENT_MANAGER_LOG)
  File "/usr/lib/python3.6/site-packages/ha/core/config/config_manager.py", line 52, in init
    ConfigManager._safe_load(const.HA_GLOBAL_INDEX, f"yaml://{const.HA_CONFIG_FILE}")
  File "/usr/lib/python3.6/site-packages/ha/core/config/config_manager.py", line 105, in _safe_load
    Conf.load(index, url)
  File "/usr/lib/python3.6/site-packages/cortx/utils/conf_store/conf_store.py", line 260, in load
    Conf._conf.load(index, url, recurse=recurse, **kwargs)
  File "/usr/lib/python3.6/site-packages/cortx/utils/conf_store/conf_store.py", line 95, in load
    kv_store = KvStoreFactory.get_instance(kvs_url, self._delim)
  File "/usr/lib/python3.6/site-packages/cortx/utils/kv_store/kv_store.py", line 134, in get_instance
    delim)
  File "/usr/lib/python3.6/site-packages/cortx/utils/kv_store/kv_store_collection.py", line 73, in __init__
    with open(self._store_path, 'w+') as f:
FileNotFoundError: [Errno 2] No such file or directory: '/etc/cortx/ha/ha.conf'
```
For this delivery, they have planned to hardcode the topic subscription (since hare is the "only" consumer). Later,  there will be REST APIs to support event subscriptions.

> re-implementing the logic that should be in cortx-ha.
Yes, indeed. 
About having a constant for topic name in cortx-ha, @ArchanaLimaye told me, they already have a constant for **topic prefix**, and they substitute the component id which hare provides. So we could avoid having a new constant and use the same.
 
 Is there any better way for handling this?